### PR TITLE
feat(docs): updated the document query and webhook sections with clarifications

### PR DESCRIPTION
**Pull request by RamilGaripov**
refs. metriport/metriport#1017

### Description

- Added clarifications on the results of the `POST /document/query` calls to relevant sections.

### Screenshots
Screenshot 2023-10-19 at 4 32 00 PM
Screenshot 2023-10-19 at 4 32 37 PM

### Release Plan

- Nothing special

### CODE REVIEW

**Review by leite08 — APPROVED** [review]

File: docs/_snippets/document-query-complete-tip.mdx [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<Tip>
+  There is a scenario in which a previous document query has already retrieved all of the available
+  documents for a patient, in which case, the `medical.document-download` webhook notification will
+  contain an empty document array: `documents: []`. Simply fetch all of the available documents with
+  [this endpoint](/medical-api/api-reference/document/get-document).
```
Comment (Goncharo) [inline docs/_snippets/document-query-complete-tip.mdx:5]: this is verbose, and at this point the dev might now have a technical understanding of the webhooks - why not just add a quick line to the existing tip? ie

```
<Tip>
  To get updates about the document query progress, including download and conversion status, check
  out our [webhooks guide](/medical-api/more-info/webhooks).
  
  Note that the webhooks will only contain updates for new data fetched in the current document query.
</Tip>
```

File: docs/medical-api/api-reference/document/start-document-query.mdx [hunk 2]
```diff
@@ -20,6 +20,10 @@ If there's no document to be converted, the total will be set to zero and the st
   out our [webhooks guide](/medical-api/more-info/webhooks).
 </Tip>
 
+Note that the webhook resulting from this query **does not** return all of the available documents for a patient - only what was retrieved with this current query.
```
Comment (Goncharo) [inline docs/medical-api/api-reference/document/start-document-query.mdx:23]: can remove this if previous comment is addressed

File: docs/medical-api/more-info/webhooks.mdx [hunk 3]
```diff
@@ -25,7 +25,10 @@ These are messages you can expect to receive in the following scenarios:
 
 1. When [queried documents](/medical-api/api-reference/document/start-document-query) have completed downloading (`type` for a patient in
    the message will be `document-download`, and at this point you'll be able to [download the raw files](/medical-api/api-reference/document/get-document));
-1. Then, if the downloaded documents contained C-CDA/XML files, when the conversion to FHIR has completed (`type` for a patient in the message
+
+<Snippet file="document-query-complete-tip.mdx" />
+
+2. Then, if the downloaded documents contained C-CDA/XML files, when the conversion to FHIR has completed (`type` for a patient in the message
    will be `document-conversion`, and at this point you'll be able to query for [patient consolidated data](#patient-consolidated-data) in FHIR-compliant format).
```
Comment (Goncharo) [inline docs/medical-api/more-info/webhooks.mdx:33]: I'd remove the other tip and just add a tip here saying something simple like:

<Tip>
  Note that the webhooks will only contain updates for new data fetched in the current document query.
</Tip>

**Review by Goncharo — CHANGES_REQUESTED** [review]
let's simplify this change - it seems to add too much cognitive overhead for what it's worth

**Pull request merged**